File: GUI/src/app/shared/models/notification/notification.ts
import { Appointment } from "../appointment/appointment";
import { User } from "../users/user";
import { v4 as uuidv4 } from 'uuid';

export class Notification {
    public id: string;
    notificationStatus: 'READ' | 'UNREAD';
    sender: User;
    recipient: User;
    title: string;
    message: string;
    sentAt: Date;
    appointment: Appointment;
    read: boolean = false;

    constructor() {
        this.id = uuidv4();
      }
}

File: CoreServices/src/main/java/com/example/entities/Notification.java
package com.example.entities;

import com.example.enums.ENotificationStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "notification")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Notification {
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @Column(columnDefinition = "BINARY(16)")
    private UUID id;
    @Enumerated(EnumType.STRING)
    private ENotificationStatus notificationStatus;
    @ManyToOne
    @JoinColumn(name = "sender_id")
    private UserEntity sender;
    @ManyToOne
    @JoinColumn(name = "recipient_id")
    private UserEntity recipient;
    private String title;
    private String message;
    private LocalDateTime sentAt;
    @ManyToOne
    @JoinColumn(name = "appointment_id", nullable = true)
    private Appointment appointment;

    @PrePersist
    protected void onCreate() {
        sentAt = LocalDateTime.now();
    }
}
